tests/Kiota.Builder.Tests/Writers/CSharp/CodeMethodWriterTests.cs

```diff
@@ -0,0 +1,289 @@
+            Assert.Contains("WriteStringValue", result);
+            Assert.Contains("WriteCollectionOfPrimitiveValues", result);
+            Assert.Contains("WriteCollectionOfObjectValues", result);
+            Assert.Contains("WriteEnumValue", result);
+            Assert.Contains("WriteAdditionalData(additionalData);", result);
+        }
+        [Fact]
+        public void WritesMethodAsyncDescription() {
+            
+            method.Description = methodDescription;
+            var parameter = new CodeParameter(method){
+                Description = paramDescription,
+                Name = paramName
+            };
+            parameter.Type = new CodeType(parameter) {
+                Name = "string"
+            };
+            method.AddParameter(parameter);
+            writer.Write(method);
+            var result = tw.ToString();
+            Assert.Contains("/// <summary>", result);
+            Assert.Contains(methodDescription, result);
+            Assert.Contains("<param name=", result);
+            Assert.Contains("</param>", result);
+            Assert.Contains(paramName, result);
+            Assert.Contains(paramDescription, result); 
+            Assert.Contains("</summary>", result);
+        }
+        [Fact]
+        public void WritesMethodSyncDescription() {
+            
+            method.Description = methodDescription;
+            method.IsAsync = false;
+            var parameter = new CodeParameter(method){
+                Description = paramDescription,
+                Name = paramName
+            };
+            parameter.Type = new CodeType(parameter) {
+                Name = "string"
+            };
+            method.AddParameter(parameter);
+            writer.Write(method);
+            var result = tw.ToString();
+            Assert.DoesNotContain("@returns a Promise of", result);
+        }
+        [Fact]
+        public void Defensive() {
+            var codeMethodWriter = new CodeMethodWriter(new CSharpConventionService());
+            Assert.Throws<ArgumentNullException>(() => codeMethodWriter.WriteCodeElement(null, writer));
+            Assert.Throws<ArgumentNullException>(() => codeMethodWriter.WriteCodeElement(method, null));
+        }
+        [Fact]
+        public void ThrowsIfParentIsNotClass() {
+            method.Parent = CodeNamespace.InitRootNamespace();
+            Assert.Throws<InvalidOperationException>(() => writer.Write(method));
+        }
+        [Fact]
+        public void ThrowsIfReturnTypeIsMissing() {
+            method.ReturnType = null;
+            Assert.Throws<InvalidOperationException>(() => writer.Write(method));
+        }
+        [Fact]
+        public void WritesReturnType() {
+            writer.Write(method);
+            var result = tw.ToString();
+            Assert.Contains(methodName, result);
+            Assert.Contains(returnTypeName, result);
+            Assert.Contains("Task<", result);// async default
```

Comment:
I suggest that we combine these three tests into a single one since we expect them to occur together in the same order every time. This will test the format as well as the occurrence. A change in this format could be an error.

`Assert.Contains($"Task<{returnTypeName}> {methodName}", result);`

We should apply this to any set of assertions testing a single line where there is a single format.

Reply:
done, the reason I'm not doing that too often is to avoid beeing too dependant on indent/spacing and also because I don't want to end up duplicating the actual code in the unit test.